### DISCUSSION TITLE
Rotate with switch

### DISCUSSION THREAD

**Discussion by caf2022** [savonet/liquidsoap#4106] [General]
Hi everyone!

At this moment I'm using rotate to play a musics and spots:

```
radio = playlist(id="playlist", reload_mode="watch", loop=true, mode="normal", "/home/playlist.pls")
spots = playlist(id="inter", reload_mode="watch", loop=true, mode="normal","/home/spots/")
radio = rotate(weights = [5,1],[spots,radio])
```
This will play 5 spots and 1 music.

How I can do the same, but with switch? to play 5 spots on each 5 minutes for example? But the 5 spots can be 1, 2, 10 I need this free to change to any number and the 5 minutes too, is this posible?

**Comment by vitoyucepi** [comment]
Hi @caf2022,
The type of the `weights` named parameter in the `random` and `rotate` functions is `[{int}]`. You can put not only an integer there, but also a function that returns an integer or a reference to an integer.
```
def a()
  1
end
b = ref(1)
c = 1
s = rotate(weights = [a, b, c], [s1, s2, s3])
```
